### PR TITLE
[Feature] Rename table property enable_storage_cache to datacache.enable

### PR DESCRIPTION
**Pull request by sduzh**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.0

File: fe/fe-core/src/main/java/com/starrocks/service/InformationSchemaDataSource.java [hunk 1]
```diff
@@ -194,8 +194,8 @@ private static Map<String, String> genProps(Table table) {
         // enable storage cache && cache ttl
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/service/InformationSchemaDataSource.java:194]: update comment

**Comment by imay** [comment]
data cache.enable is better because existed variables have the name with enable other than enabled. Better to keep consistent with them

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 2]
```diff
@@ -941,7 +941,7 @@ public static List<ForeignKeyConstraint> analyzeForeignKeyConstraint(
     // analyze enable storage cache and cache ttl, and whether allow async write back
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:941]: update comment

**Review by wyb — DISMISSED** [review]

**Comment by sduzh** [comment]
> data cache.enable is better because existed variables have the name with enable other than enabled. Better to keep consistent with them

By convention, boolean config options typically use "enabled" rather than just "enable". This is a more self-documenting name and clearly indicates it is a boolean flag.

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 68 (79.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|0|1|00.00%|[395]|
|:large_blue_circle:|com/starrocks/lake/backup/LakeRestoreJob.java|0|2|00.00%|[301, 316]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|[2582, 2584]|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|0|2|00.00%|[196, 197]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|4|6|66.67%|[703, 704]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|3|66.67%|[1686]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|12|66.67%|[1396, 4040, 4041, 4513]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/SinglePartitionPersistInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/DataCacheInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StorageInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SinglePartitionDesc.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/PartitionPersistInfoV2.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/RangePartitionPersistInfo.java|1|1|100.00%|[]|

**Comment by sduzh** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 68 (79.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|0|1|00.00%|[395]|
|:large_blue_circle:|com/starrocks/lake/backup/LakeRestoreJob.java|0|2|00.00%|[301, 316]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|[2582, 2584]|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|0|2|00.00%|[196, 197]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|4|6|66.67%|[703, 704]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|3|66.67%|[1686]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|12|66.67%|[1396, 4040, 4041, 4513]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/SinglePartitionPersistInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/DataCacheInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ListPartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StorageInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SinglePartitionDesc.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/PartitionPersistInfoV2.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/RangePartitionPersistInfo.java|1|1|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**